File: android/app/src/main/java/com/transcribecare/app/viewmodel/SettingsViewModel.kt
package com.transcribecare.app.viewmodel

import android.app.Application
import android.content.Context
import androidx.lifecycle.AndroidViewModel
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.asStateFlow

/**
 * ViewModel for the Settings screen managing user preferences.
 * Persists settings to SharedPreferences so they survive app restarts.
 */
class SettingsViewModel(application: Application) : AndroidViewModel(application) {

    private val prefs = application.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)

    private val _largeTextMode = MutableStateFlow(
        prefs.getBoolean(KEY_LARGE_TEXT_MODE, true)
    )
    val largeTextMode: StateFlow<Boolean> = _largeTextMode.asStateFlow()

    /**
     * Toggles the large text mode preference and persists the new value.
     */
    fun toggleLargeTextMode() {
        val newValue = !_largeTextMode.value
        _largeTextMode.value = newValue
        prefs.edit().putBoolean(KEY_LARGE_TEXT_MODE, newValue).apply()
    }

    /**
     * Sets the large text mode preference to a specific value and persists it.
     */
    fun setLargeTextMode(enabled: Boolean) {
        _largeTextMode.value = enabled
        prefs.edit().putBoolean(KEY_LARGE_TEXT_MODE, enabled).apply()
    }

    companion object {
        private const val PREFS_NAME = "transcribecare_settings"
        private const val KEY_LARGE_TEXT_MODE = "large_text_mode"
    }
}

File: ios/TranscribeCare/ViewModel/SettingsViewModel.swift
import Foundation

/// ViewModel for the Settings screen managing user preferences.
/// Persists the Large Text Mode preference using UserDefaults.
@Observable
class SettingsViewModel {

    /// UserDefaults key for the Large Text Mode preference.
    static let largeTextModeKey = "largeTextModeEnabled"

    /// Whether Large Text Mode is enabled (36pt minimum for transcript text).
    /// Backed by UserDefaults for persistence across app launches.
    var largeTextMode: Bool {
        didSet {
            UserDefaults.standard.set(largeTextMode, forKey: Self.largeTextModeKey)
        }
    }

    init() {
        self.largeTextMode = UserDefaults.standard.bool(forKey: Self.largeTextModeKey)
    }

    /// Toggles the Large Text Mode preference.
    func toggleLargeTextMode() {
        largeTextMode.toggle()
    }
}
